internal/migrate/versions.go

```diff
@@ -90,6 +97,29 @@ func (migration *Migration) ValidateSteps() error {
 	return nil
 }
 
+// ValidateVersions checks that the version of the migration matches the state of the database
+func (migration *Migration) ValidateVersions(log *zap.Logger) error {
+	for _, step := range migration.Steps {
+		dbVersion, err := migration.getLatestVersion(log, step.DB)
+		if err != nil {
+			return ErrValidateVersionQuery.Wrap(err)
+		}
+
+		if step.Version > dbVersion {
+			return ErrValidateVersionMismatch.New("expected %d <= %d", step.Version, dbVersion)
```

Comment:
Suggest:

    "expected %T db %+v to be at version %d or greater, but found version %d", step.DB, step.DB, step.Version, dbVersion

Just cause it might be hard to understand a message like `validate db version mismatch error: expected 5 <= 8`. If I saw that, I'd be like "ok but 5 IS less than 8..."